src/economy.cpp

```diff
@@ -1612,7 +1613,8 @@ static void ReserveConsist(Station *st, Vehicle *u, CargoArray *consist_capleft,
  */
 static void UpdateLoadUnloadTicks(Vehicle *front, const Station *st, int ticks)
 {
-	if (front->type == VEH_TRAIN) {
+	if (_cheats.loading_speed.value) {}		
+	else if (front->type == VEH_TRAIN) {
```

Comment:
Try this instead of the `if`/`else if` silliness. 😄 
```suggestion
	if (front->type == VEH_TRAIN && !_cheats.loading_speed.value) {
```

While I'm at it, a little C++ lesson for you:

We use `&&` (logical AND) and `||` (logical OR) a lot in OpenTTD. These, and using [early returns](https://softwareengineering.stackexchange.com/questions/18454/should-i-return-from-a-function-early-or-use-an-if-statement) instead of `if`/`else` blocks, are two of the simplest and most common tools you should get used to reaching for in your toolbelt.

It's worth noting that these logical operators are short-circuiting: in my suggestion, if `front->type == VEH_TRAIN` is false, we skip past the block without even checking the cheat value. The opposite is true for `||`, where if the first condition is true we enter the block without checking the second condition. 

In this case, the order of the conditions doesn't matter. But sometimes they do, like if we're checking `IsThingValid() && ValueOfThing() == stuff` where trying to get `ValueOfThing()` of an invalid thing would cause a crash. The short-circuit makes it safe. The other use for choosing a specific order is if one check is computationally cheap and another is more expensive; we can avoid checking the expensive condition unnecessarily by placing it second.

Reply:
I almost had it like that, I just used a comma instead of && because I'm used to NML (it obviously didn't work) :)